pkg/sql/types/types.go

```diff
@@ -212,13 +213,40 @@ func (e *EnumMetadata) debugString() string {
 	)
 }
 
-// UserDefinedTypeName is an interface for the types package to manipulate
-// qualified type names from higher level packages for name display.
-type UserDefinedTypeName interface {
-	// Basename returns the unqualified name of the type.
-	Basename() string
-	// FQName returns the fully qualified name of the type.
-	FQName() string
+// UserDefinedTypeName is a struct representing a qualified user defined
+// type name. We redefine a common struct from higher level packages. We
+// do so because proto will panic if any members of a proto struct are
+// private. Rather than expose private members of higher level packages,
+// we define a separate type here to be safe.
+type UserDefinedTypeName struct {
+	Catalog string
```

Comment:
should this be inheriting / implement the `ObjectName` interface?

Reply:
Using interfaces seems like the cause of the problem -- because something with unexported fields could implement the interface, we could have that problem in the future. I think its safest to just have an explicit struct here.